application/src/main/java/org/thingsboard/server/service/security/auth/TokenOutdatingService.java

```diff
@@ -49,34 +51,51 @@ protected void initCache() {
 
     @EventListener(classes = UserAuthDataChangedEvent.class)
     public void onUserAuthDataChanged(UserAuthDataChangedEvent userAuthDataChangedEvent) {
-        outdateOldUserTokens(userAuthDataChangedEvent.getUserId());
+        outdateTokens(userAuthDataChangedEvent.getUserId().getId().toString());
     }
 
     public boolean isOutdated(JwtToken token, UserId userId) {
         Claims claims = tokenFactory.parseTokenClaims(token).getBody();
         long issueTime = claims.getIssuedAt().getTime();
 
-        return Optional.ofNullable(tokenOutdatageTimeCache.get(toKey(userId), Long.class))
-                .map(outdatageTime -> {
-                    if (System.currentTimeMillis() - outdatageTime <= SECONDS.toMillis(jwtSettings.getRefreshTokenExpTime())) {
-                        return MILLISECONDS.toSeconds(issueTime) < MILLISECONDS.toSeconds(outdatageTime);
-                    } else {
-                        /*
-                         * Means that since the outdating has passed more than
-                         * the lifetime of refresh token (the longest lived)
-                         * and there is no need to store outdatage time anymore
-                         * as all the tokens issued before the outdatage time
-                         * are now expired by themselves
-                         * */
-                        tokenOutdatageTimeCache.evict(toKey(userId));
-                        return false;
-                    }
-                })
-                .orElse(false);
+        Jws<Claims> claimsJws = tokenFactory.parseTokenClaims(token);
+        Optional<String> tokenId = Optional.ofNullable(claimsJws.getBody().get("tokenId").toString());
```

Comment:
`claimsJws.getBody().get("tokenId")` might be null, not `claimsJws.getBody().get("tokenId").toString()`

Reply:
resolved